apps/dash-uber-rides-demo/app.py

```diff
@@ -0,0 +1,591 @@
+                        [0.4583, "#28C86D"],
+                        [0.50, "#29C481"],
+                        [0.54167, "#2AC093"],
+                        [0.5833, "#2BBCA4"],
+                        [1.0, "#613099"],
+                    ],
+                    colorbar=dict(
+                        title="Time of<br>Day",
+                        x=0.93,
+                        xpad=0,
+                        nticks=24,
+                        tickfont=dict(color="#d8d8d8"),
+                        titlefont=dict(color="#d8d8d8"),
+                        thicknessmode="pixels",
+                    ),
+                ),
+            ),
+            # Plot of important locations on the map
+            Scattermapbox(
+                lat=[list_of_locations[i]["lat"] for i in list_of_locations],
+                lon=[list_of_locations[i]["lon"] for i in list_of_locations],
+                mode="markers",
+                hoverinfo="text",
+                text=[i for i in list_of_locations],
+                marker=dict(size=8, color="#ffa0a0"),
+            ),
+        ],
+        layout=Layout(
+            autosize=True,
+            height=500,
+            margin=go.layout.Margin(l=0, r=0, t=0, b=0),
+            showlegend=False,
+            mapbox=dict(
+                accesstoken=mapbox_access_token,
+                center=dict(lat=latInitial, lon=lonInitial),  # 40.7272  # -73.991251
+                style="dark",
+                bearing=bearing,
+                zoom=zoom,
+            ),
+            updatemenus=[
+                dict(
+                    buttons=(
+                        [
+                            dict(
+                                args=[
+                                    {
+                                        "mapbox.zoom": 12,
+                                        "mapbox.center.lon": "-73.991251",
+                                        "mapbox.center.lat": "40.7272",
+                                        "mapbox.bearing": 0,
+                                        "mapbox.style": "dark",
+                                    }
+                                ],
+                                label="Reset Zoom",
+                                method="relayout",
+                            )
+                        ]
+                    ),
+                    direction="left",
+                    pad={"r": 0, "t": 0, "b": 0, "l": 0},
+                    showactive=False,
+                    type="buttons",
+                    x=0.45,
+                    y=0.02,
+                    xanchor="left",
+                    yanchor="bottom",
+                    bgcolor="#323130",
+                    borderwidth=1,
+                    bordercolor="#6d6d6d",
+                    font=dict(color="#FFFFFF"),
+                )
+            ],
+        ),
+    )
+
+
+@app.server.before_first_request
+def defineTotalList():
+    global totalList
```

Comment:
Please don't use global variables! I would use a `dcc.Store` instead. (Unless it's vital to have a global variable in the app for some reason?)

Reply:
I don't think the global variable is necessary. It was just a hold over from the previous app design.